vendor/github.com/TheWeatherCompany/softlayer-go/services/softlayer_network_application_delivery_controller.go

```diff
@@ -272,20 +274,68 @@ func (slnadcs *softLayer_Network_Application_Delivery_Controller_Service) GetObj
 	return nadc, nil
 }
 
+func (slnadcs *softLayer_Network_Application_Delivery_Controller_Service) GetBillingItem(volumeId int) (datatypes.SoftLayer_Billing_Item, error) {
+
+	response, errorCode, err := slnadcs.client.GetHttpClient().DoRawHttpRequest(fmt.Sprintf("%s/%d/getBillingItem.json", slnadcs.GetName(), volumeId), "GET", new(bytes.Buffer))
+	if err != nil {
+		return datatypes.SoftLayer_Billing_Item{}, err
+	}
+
+	if common.IsHttpErrorCode(errorCode) {
+		errorMessage := fmt.Sprintf("softlayer-go: could not SoftLayer_NetWork_Storage#getBillingItem, HTTP error code: '%d'", errorCode)
+		return datatypes.SoftLayer_Billing_Item{}, errors.New(errorMessage)
+	}
+
+	billingItem := datatypes.SoftLayer_Billing_Item{}
+	err = json.Unmarshal(response, &billingItem)
+	if err != nil {
+		return datatypes.SoftLayer_Billing_Item{}, err
+	}
+
+	return billingItem, nil
+}
+
 func (slnadcs *softLayer_Network_Application_Delivery_Controller_Service) DeleteObject(id int) (bool, error) {
-	response, errorCode, err := slnadcs.client.GetHttpClient().DoRawHttpRequest(fmt.Sprintf("%s/%d.json", slnadcs.GetName(), id), "DELETE", new(bytes.Buffer))
+	billingItem, err := slnadcs.GetBillingItem(id)
 	if err != nil {
-		errorMessage := fmt.Sprintf("softlayer-go: could not perform SoftLayer_Network_Application_Delivery_Controller#deleteObject, error message '%s'", err.Error())
-		return false, errors.New(errorMessage)
+		return false, err
+	}
+
+	if billingItem.Id > 0 {
+		billingItemService, err := slnadcs.client.GetSoftLayer_Billing_Item_Service()
+		if err != nil {
+			return false, err
+		}
+
+		deleted, err := billingItemService.CancelService(billingItem.Id)
+		if err != nil {
+			return false, err
+		}
+
+		if deleted {
+			return false, nil
+		}
+	}
+
+	fmt.Errorf("softlayer-go: could not SoftLayer_Network_Storage_Service#deleteIscsiVolume with id: '%d'", id)
+
+	return true, err
+}
```

Comment:
Do we need `bool, error` as the return signature for DeleteObject? or would just `error` suffice?